lib/handler/cache-handler.js

```diff
@@ -69,15 +69,19 @@ class CacheHandler extends DecoratorHandler {
       statusCode <= 399
     ) {
       // https://www.rfc-editor.org/rfc/rfc9111.html#name-invalidating-stored-respons
-      const result = this.#opts.store.deleteByOrigin(this.#req.origin)
-      if (
-        result &&
-        typeof result.catch === 'function' &&
-        typeof this.#handler.onError === 'function'
-      ) {
-        result.catch(err => {
-          process.emitWarning(`deleteByOrigin failed: ${err}`)
-        })
+      // Try/catch for if it's synchronous
+      try {
+        const result = this.#opts.store.deleteByOrigin(this.#req.origin)
+        if (
+          result &&
+          typeof result.catch === 'function' &&
+          typeof this.#handler.onError === 'function'
+        ) {
+          // Fail silently
+          result.catch(_ => {})
+        }
+      } catch (err) {
+        // Fail silently
```

Comment:
shouldn't we call on error here?

Reply:
re https://github.com/nodejs/undici/pull/3562#discussion_r1791277840